File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>


int main(int argc, string argv[])
{
    int i, j, c, k, p;
    if (argc != 2)
    {
        printf("Wrong command-line argument");
        return 1;
    }
    k = atoi(argv[1]);
    string plaintext = get_string("plaintext:");
    for (i = 0, j = strlen(plaintext); i < j; i++)
    {
        c = plaintext[i];
        if (isalpha(c) == 0)
        {
            plaintext[i] = c;
            continue;
        }
        if (c >= 65 && c <= 90)
        {
            c = (c - 65);
            p = 65;
        }
        else if (c >= 97 && c <= 122)
        {
            c = (c - 97);
            p = 97;
        }
        c = (c + k) % 26;
        plaintext[i] = (c + p);
    }
    printf("ciphertext: %s", plaintext);
    printf("\n");
    return 0;
}

File: pset6/caesar.py
import sys
# check if it have proper cmnd line argument
# or exit
if len(sys.argv) != 2:
    sys.exit(1)
plainText = input('plain_text:')
# store the key value to encrypt
key = int(sys.argv[1])
textList = list(plainText)
for index, char in enumerate(textList):
    ascii = ord(char)
    # if not alpabetic char, ignore
    if not char.isalpha():
        continue
    # c -> position relative to 65 or 97
    # p -> relativr base
    if ascii >= 65 and ascii <= 90:
        c = ascii - 65
        p = 65
    elif ascii >= 97 and ascii <= 122:
        c = ascii - 97
        p = 97
    c = (c + key) % 26
    # chang the char in place
    textList[index] = chr(c + p)
print('ciphertext: ', ''.join(textList))